src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -336,6 +334,76 @@ int64_t ClusterResourceScheduler::GetBestSchedulableNode(const TaskRequest &task
   return best_node;
 }
 
+int64_t ClusterResourceScheduler::GetBestSchedulableNode(const TaskRequest &task_req,
+                                                         bool actor_creation,
+                                                         bool force_spillback,
+                                                         int64_t *total_violations,
+                                                         bool *is_infeasible) {
+  if (!hybrid_spillback_) {
+    return GetBestSchedulableNodeLegacy(task_req, actor_creation, force_spillback,
+                                        total_violations, is_infeasible);
+  }
+
+  std::vector<int64_t> round;
+  {
+    round.push_back(local_node_id_);
+    for (const auto &pair : nodes_) {
+      if (pair.first != local_node_id_) {
+        round.push_back(pair.first);
+      }
+    }
+    auto begin = round.begin();
+    // Make the local node the first item, and skip it when sorting.
+    begin++;
+    std::sort(begin, round.end());
+  }
+
+  int64_t best_node_id = -1;
+  float best_utilization_score = 1.0;
+  bool best_is_available = false;
+
+  for (auto node_id : round) {
```

Comment:
Can you add unit tests for this method? It would be best to split it out into a standalone function that can be tested independently of the entire class.